packages/oui-pagination/src/pagination.controller.js

```diff
@@ -0,0 +1,76 @@
+export default class {
+    constructor (ouiPaginationConfiguration) {
+        "ngInject";
+
+        this.config = ouiPaginationConfiguration;
+        this.pageSizeList = this.config.pageSizeList;
+    }
+
+    $onChanges (changes) {
+        // If pageSizeMax changes, recalculate pageSizeList...
+        if (changes.pageSizeMax && this.pageSizeMax) {
+            this.pageSizeList = this.config.pageSizeList
+                .sort((a, b) => a - b)
+                .filter(pageSize => pageSize <= this.pageSizeMax);
+
+            // ... and add the max page size if necessary.
+            if (this.pageSizeList.indexOf(this.pageSizeMax) < 0) {
+                this.pageSizeList.push(this.pageSizeMax);
+            }
+        }
+
+        // Instead of recalculate these values on each digest,
+        // values are calculated on each change concerning the table.
+        // It avoids digest problems in ng-repeat.
+        this.pageCount = this.getPageCount();
+        this.pageRange = this.getPageRange();
+        this.currentPage = this.getCurrentPage();
+
+        this.processTranslations();
+    }
+
+    processTranslations () {
+        this.translations = Object.assign({}, this.config.translations);
+        this.translations.ofNResults = this.translations.ofNResults
+            .replace("{{totalItems}}", this.totalItems);
+        this.translations.currentPageOfPageCount = this.translations.currentPageOfPageCount
+            .replace("{{currentPage}}", this.currentPage)
+            .replace("{{pageCount}}", this.pageCount);
+    }
+
+    _onChange () {
+        this.onChange({
+            $event: {
+                offset: this.currentOffset,
+                pageSize: this.pageSize
+            }
+        });
+    }
+
+    onPageSizeChange (pageSize) {
+        this.pageSize = pageSize;
+        this._onChange();
+    }
+
+    onPageChange (page) {
+        this.currentOffset = (this.pageSize * (page - 1)) + 1;
+        this._onChange();
+    }
+
+    getLastItemOffset () {
+        return Math.min(this.currentOffset + this.pageSize - 1, this.totalItems);
+    }
+
+    getPageCount () {
+        return Math.ceil(this.totalItems / this.pageSize);
+    }
+
+    getCurrentPage () {
+        return Math.floor((this.currentOffset - 1) / this.pageSize) + 1;
+    }
+
+    getPageRange () {
+        return Array(...{ length: this.getPageCount() })
```

Comment:
No, not the same thing.

Reply:
Ok, I confirm it is not the same thing, it produce an array with the right number of item, but they can't be used. We need to execute the `fill` method after, otherwise we can't iterator over those elements. 

`Array(5).map((value, index) => index + 1)` produce `[ , , , , ]` instead of `[ 1, 2, 3, 4, 5]`

So, `Array(this.getPageCount()).fill().map(...)` is really equivalent of what you have done. I still don't really know why `...{ length: [...] }` seems to work, but it is OK for me (didn't found any documentation about it).